lib/route-recognizer.ts

```diff
@@ -447,11 +450,17 @@ function findHandler(state: State, originalPath: string, queryParams: QueryParam
       }
     }
 
-    result[i] = {
+    let compiledResult: Result = {
       handler: handler.handler,
       params,
       isDynamic
     };
+
+    if (handler.metadata) {
```

Comment:
We should likely always have `metadata` in the compiled result. There are some performance implications here of introducing a new slot onto an object post creation.